### PR TITLE
Frontend

### PR DESCRIPTION
**Pull request by samysellami**

### CODE REVIEW

File: frontend/frontend/src/App.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+function App() {
+    return (
+        <BrowserRouter>
+            <Header />
+            <main className='py-3'>
+                <Container>
+                    <Routes>
+                        <Route exact path='/' element={<HomeScreen />} />
+                        <Route path='/expense/:id' element={<ExpenseScreen />} />
```
Comment (workshur) [inline frontend/frontend/src/App.js]: unused route

File: frontend/frontend/src/screens/ExpenseScreen.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import React from 'react'
+
+const ExpenseScreen = () => {
+    return <div></div>
+}
+
+export default ExpenseScreen
```
Comment (workshur) [inline frontend/frontend/src/screens/ExpenseScreen.js]: this component renders nothing

File: frontend/frontend/src/components/Expense.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+                <Card.Text as='div'>
+                    <div className='my-3'>
+                        <strong>{request.description}</strong>
+                    </div>
+                </Card.Text>
+            </Card.Body>
+        </Card>
+    )
+}
+
```
Comment (workshur) [inline frontend/frontend/src/components/Expense.js]: I couldn't find a place where we use this component

File: frontend/frontend/src/screens/HomeScreen.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
+import React, { useEffect, useState } from 'react'
+import { Form, Button, Container, Row, Col } from 'react-bootstrap'
+import { Card } from 'react-bootstrap'
+import { Link } from 'react-router-dom'
+
+import axios from 'axios'
+import Message from '../components/Message'
+import Loader from '../components/Loader'
```
Comment (workshur) [inline frontend/frontend/src/screens/HomeScreen.js]: I couldn't find a place where we use Message, Loader and axios

File: frontend/frontend/src/screens/HomeScreen.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+function HomeScreen() {
+    useEffect(() => {}, [])
```
Comment (workshur) [inline frontend/frontend/src/screens/HomeScreen.js]: Could you describe what this hooks call is for?

File: frontend/frontend/src/screens/ListExpensesScreen.js [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+            {errorDelete && <Message variant='danger'>{errorDelete}</Message>}
+
+            {loadingCreate && <Loader />}
+            {errorCreate && <Message variant='danger'>{errorCreate}</Message>}
+
+            {loading ? (
+                <Loader />
+            ) : error ? (
+                <Message variant='danger'>{error}</Message>
+            ) : (
```
Comment (workshur) [inline frontend/frontend/src/screens/ListExpensesScreen.js:152]: Is there any way to organize all these Loaders and messages and make it more "DRY".

File: frontend/frontend/src/screens/ChartScreen.js [hunk 7]
```diff
@@ -0,0 +1,210 @@
...
+                        expense.category == category._id &&
+                        months[Number(expense.date.substring(5, 7))] === month
+                    ) {
+                        return acc + expense.amount
+                    } else {
+                        return acc
+                    }
+                }, 0)
+            )
+        })
```
Comment (workshur) [inline frontend/frontend/src/screens/ChartScreen.js:108]: Too complex

File: frontend/frontend/src/App.js (hunk 1 above)
Comment (samysellami) [reply frontend/frontend/src/App.js]: Yes, forgot to remove it (, I will do it

File: frontend/frontend/src/screens/ExpenseScreen.js (hunk 2 above)
Comment (samysellami) [reply frontend/frontend/src/screens/ExpenseScreen.js]: I will remove it

File: frontend/frontend/src/components/Expense.js (hunk 3 above)
Comment (samysellami) [reply frontend/frontend/src/components/Expense.js]: True, this is a collateral damage of my first attempts, will remove it

File: frontend/frontend/src/screens/HomeScreen.js (hunk 4 above)
Comment (samysellami) [reply frontend/frontend/src/screens/HomeScreen.js]: Yes, I will remove them

File: frontend/frontend/src/screens/HomeScreen.js (hunk 5 above)
Comment (samysellami) [reply frontend/frontend/src/screens/HomeScreen.js]: For nothing actually, this a typo that will be fixed

File: frontend/frontend/src/screens/ListExpensesScreen.js (hunk 6 above)
Comment (samysellami) [reply frontend/frontend/src/screens/ListExpensesScreen.js:152]: There should be for sure, but I think that it is better to separate the delete, create and list loadings and messages here

File: frontend/frontend/src/screens/ChartScreen.js (hunk 7 above)
Comment (samysellami) [reply frontend/frontend/src/screens/ChartScreen.js:108]: This part is related to the table that can be removed because it is not a requirement

Comment (workshur) [reply frontend/frontend/src/screens/ChartScreen.js:108]: Are you sure you use correct iteration method here?

Comment (samysellami) [reply frontend/frontend/src/screens/ChartScreen.js:108]: Do you mean that the results are incorrect or that the iteration method is not suitable (efficient)?

**Pull request closed without merging**